constructor/construct.py

```diff
@@ -701,53 +962,57 @@ def verify(info):
         types_key[key] = types
         if required:
             required_keys.add(key)
-        if 'Obsolete' in descr:
+        if "Obsolete" in descr:
             obsolete_keys.add(key)
 
     for key in info:
         if key not in types_key:
             sys.exit("Error: unknown key '%s' in construct.yaml" % key)
         elt = info[key]
         if key in obsolete_keys:
-            sys.stderr.write("Warning: key '%s' is obsolete.\n"
-                             "  Its value '%s' is being ignored.\n" % (key, elt))
+            sys.stderr.write(
+                "Warning: key '%s' is obsolete.\n"
+                "  Its value '%s' is being ignored.\n" % (key, elt)
+            )
         types = types_key[key]
         if not isinstance(elt, types):
-            sys.exit("Error: key '%s' points to %s,\n"
-                     "       expected %s" % (key, type(elt), types))
+            sys.exit(
```

Comment:
Another "unrelated but":

Exception handling and `sys.exit` are used interchangeably and should be revisited.

Reply:
agreed ... the exception handler can then also convert different exceptions into different exit codes